### PR TITLE
Add AMQP check

### PR DESCRIPTION
**Pull request by pobed2**
Uses [Pika](http://pika.readthedocs.io/en/0.10.0/) to check AMQP connections.

### CODE REVIEW

File: watchman/checks.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 from django.core.mail import EmailMessage
 from django.db import connections
 
+import pika
```
Comment (mwarkentin) [inline watchman/checks.py]: My main concern is the additional requirement of another library here - especially for users which aren't using rabbitmq, or are using another library.

I have two ideas that I think I would be ok with:

* Make this check optional, including this import (eg. `try ... except ImportError` and handle things smartly)
* Split this check out into a separate package (`django-watchman-amqp`?) which makes it easy to install on demand for apps which need it

@pobed2 what do you think?

File: watchman/checks.py [hunk 2]
```diff
@@ -81,3 +95,7 @@ def email():
 
 def storage():
     return {"storage": _check_storage()}
+
+
+def amqp():
+    return {'amqp': _check_ampq(settings.AMQP_CONNECTIONS)}
```
Comment (mwarkentin) [inline watchman/checks.py]: Is this a "standard" way of configuring amqp in Django, or is this specific to how we do it? That would push me towards a separate package as well..

File: watchman/checks.py [hunk 3]
```diff
@@ -67,6 +73,14 @@ def _check_storage():
     return {"ok": True}
 
 
+@check
+def _check_amqp_connection(connection_url):
+    conn_params = pika.URLParameters(connection_url)
+    connection = pika.BlockingConnection(conn_params)
+    connection.close()
+    return {'ok': True}
```
Comment (ravangen) [inline watchman/checks.py]: Can we use some identifier of the connection similar to what cache and database checks do? A possible candidate would be the `vhost` name.

This file seems to favour double quotes

File: watchman/checks.py (hunk 1 above)
Comment (ravangen) [inline watchman/checks.py]: I would suggest the optional import. The check wouldn't be used anyways if this package or some other equivalent mechanism was available manage the connection.

File: watchman/checks.py (hunk 2 above)
Comment (ravangen) [inline watchman/checks.py]: There is no standard, but we talked offline 6 months ago about using this type of convention following along with how DATABASES and CACHES are defined.

Comment (mwarkentin) [inline watchman/checks.py]: `_check_ampq` doesn't exist? Should be `_check_amqp_connections`?

File: watchman/checks.py (hunk 3 above)
Comment (mwarkentin) [inline watchman/checks.py]: +1 - if this supports multiple AMQP connections, there should be some way of telling them apart.

File: watchman/checks.py (hunk 2 above)
Comment (mwarkentin) [inline watchman/checks.py]: The main thing that's an issue to me is.. how many users of this package would actually be able to use this check and get any value out of it? My guess is very low - which pushes me towards a separate package for this.

Comment (ravangen) [inline watchman/checks.py]: I am not sure about the composition of users of this package, but you had created https://github.com/mwarkentin/django-watchman/issues/21 to do this every thing and was implemented to a similar net effect in https://github.com/mwarkentin/django-watchman/issues/77#issuecomment-215202087

File: watchman/checks.py [hunk 4]
```diff
@@ -67,6 +71,16 @@ def _check_storage():
     return {"ok": True}
 
 
+@check
+def _check_amqp_connection(connection_url):
+    import pika
```
Comment (pobed2) [inline watchman/checks.py]: Does this count as an optional import?

**Comment by mwarkentin** [comment]
@pobed2 I believe we've either pulled this check into the app as a custom check or separate library, correct? Going to close this.

**Pull request closed without merging**